### PR TITLE
Make SnackBar animate itself

### PR DESCRIPTION
**Pull request by mpcomplete**
This introduces a bit of a regression - the FloatingActionButton no longer animates. I'm not sure yet how to animated the FAB along with the SnackBar. Maybe some notion of anchoring a node to another.

Also removes the last use of AnimationBuilder.

### CODE REVIEW

File: sky/sdk/lib/widgets/drawer.dart [hunk 1]
```diff
@@ -44,6 +45,16 @@ enum DrawerStatus {
...
 
+// TODO(mpcomplete): find a better place for this.
+class AnimatedColor extends AnimatedType<Color> {
+  AnimatedColor(Color begin, { Color end, Curve curve: linear })
+    : super(begin, end: end, curve: curve);
+
+  void setFraction(double t) {
+    value = lerpColor(begin, end, t);
+  }
+}
```
Comment (abarth) [inline sky/sdk/lib/widgets/drawer.dart]: Maybe in the same file as AnimatedType?

File: sky/sdk/lib/widgets/snack_bar.dart [hunk 2]
```diff
@@ -29,18 +35,58 @@ class SnackBarAction extends Component {
...
-class SnackBar extends Component {
+class SnackBar extends AnimatedComponent {
 
   SnackBar({
     String key,
     this.content,
-    this.actions
+    this.actions,
+    this.showing,
+    this.onHidden
```
Comment (abarth) [inline sky/sdk/lib/widgets/snack_bar.dart]: In Drawer and PopupMenu, we introduced an enum to differentiate this state more clearly from `showing`.

**Comment by abarth** [comment]
This looks pretty good.  Some minor comments above.  Also, please squash your comments before merging.  LGTM w/ the change to use an enum and an onStatusChanged callback to match the idiom in Drawer and PopupMenu.  (If you don't like the idiom, we should change all three to match in a separate CL.)

File: sky/sdk/lib/widgets/drawer.dart (hunk 1 above)
Comment (mpcomplete) [inline sky/sdk/lib/widgets/drawer.dart]: I'm going to change this soon.

**Comment by mpcomplete** [comment]
Had to create a new branch after squashing the commits. https://github.com/domokit/sky_engine/pull/125

**Pull request closed without merging**